src/packet/literal_data.rs

```diff
@@ -84,6 +84,24 @@ impl LiteralData {
         &self.data
     }
 
+    #[inline]
+    /// Extracts data in to raw data
+    pub fn into_bytes(self) -> Vec<u8> {
+        self.data
+    }
+
+    #[inline]
+    /// Extracts data as string, returning raw bytes as Err if not valid utf-8 string
+    pub fn try_into_string(self) -> Result<String, Vec<u8>> {
+        match self.mode {
+            DataMode::Binary => Err(self.data),
```

Comment:
From the point of view of OpenPGP semantics, I tend to think that DataMode::Binary should also be handled like the _ case (i.e.: I think the match should be dropped).

As far as I can tell, the DataMode of a literal packet isn't very intimately linked with the expected content of the packet. So if we're willing to try String::from_utf8 for DataMode::Binary, I think we should do for all cases.

Reply:
I followed `to_string` implementation which ignores Binary variant 🤔

Reply:
Hm, I guess that's fair enough. Sorry, ignore this point :)